llvm/lib/IR/Function.cpp

```diff
@@ -1491,7 +1491,19 @@ bool Intrinsic::isConstrainedFPIntrinsic(ID QID) {
 #define INSTRUCTION(NAME, NARG, ROUND_MODE, INTRINSIC)                         \
   case Intrinsic::INTRINSIC:
 #include "llvm/IR/ConstrainedOps.def"
+#undef INSTRUCTION
     return true;
+  default:
+    return false;
+  }
+}
+
+bool Intrinsic::hasConstrainedFPRoundingModeOperand(Intrinsic::ID QID) {
```

Comment:
We should probably use SearchableTabless and generate this function

Reply:
I'd rather hold off of that for now but it does feed into my next step of adding support for target intrinsics.  

Currently I'm pinning everything on `IntrStrictFP`, which is perhaps a mistake and I should be more explicit and introduce dedicated `IntrhasRoundMetadata` and `IntrHasFPExpectMetadata` properties?

If acceptable I can follow this patch with that refactoring? I'm already planning to generate the case blocks so most uses of ConstrainedOps.def at the IR level is removed.

Reply:
I think the whole approach of having dedicated strict intrinsics with dedicated operands is a mistake, see https://discourse.llvm.org/t/thought-on-strictfp-support/71453/9

I think these would be better as some kind of optional tag on call or an operand bundle or something

Reply:
Thanks for highlighting this.  I'll reassess my next steps to see if I can do something better for target intrinsics rather than repeating the same mistakes as with the existing constrained ones.  I had naively assumed the design had settled.

Reply:
> I think these would be better as some kind of optional tag on call or an operand bundle or something

Playing around with an implementation of this has long been on a TODO list of mine, but I've not yet had time to try it out.